src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -1695,28 +1737,18 @@ internal void PerformUserCancellation()
                 {
                     if (cancellationTimeout > 0)
                     {
-                        lock (this)
```

Comment:
What prevents Break from being just after we release SingleUseLock above? Wouldn't it race with this method, possibly causing exceptions as ReadBuffer.Cts is accessed after disposal?

Reply:
It can happen, but at most it's going to change the state to `Broken`. After that it's going to wait for the cancellation to complete by `lock (CancelLock) { }`

Reply:
OK, makes sense - though as below, it isn't super obvious, is all.